### PR TITLE
Absolute imports

### PR DESCRIPTION
**Pull request by bucktower**
* Allow for absolute imports in linting without errors or warnings
  * `import { Something } from 'src/constants/specials';` instead of only allowing `import { Something } from '../../src/constants/specials';`
* Update `package.json` to require end-user to install required linting dependencies such as `eslint`, `eslint-import-resolver-typescript`, which will be more important in newer repos like `rac-web-silverware`

### CODE REVIEW

**Review by stu-k — COMMENTED** [review]
Does this repo not have any build jobs? How will we know if a PRs change is horribly breaking? Does it just publish its contents regardless? That doesn't seem safe.
Also I would think a minor version bump (0.1.0) would be for added functionality, not minor config changes.

**Comment by bucktower** [comment]
> Does this repo not have any build jobs? How will we know if a PRs change is horribly breaking? Does it just publish its contents regardless? That doesn't seem safe.
> Also I would think a minor version bump (0.1.0) would be for added functionality, not minor config changes.

@stuartkilgore I figured being able to detect absolute imports would be a feature for a linting repo, I can make it a patch version increment though if you think it's place is more of a bug fix.

There are no build jobs currently apart from publishing. I want to fix the existing linting errors in production repos first.

**Comment by stu-k** [comment]
@bucktower If the code is somehow broken, where will we find out? Here, or when they're imported/used in other apps?

**Comment by bucktower** [comment]
> @bucktower If the code is somehow broken, where will we find out? Here, or when they're imported/used in other apps?

@stuartkilgore Added in linting and CircleCI according to the stricter, vanilla Airbnb rules.

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ jobs:
       - checkout
       - attach_workspace:
           at: ~/constable
+      - run:
+        name: Linting
+        command: |
+          npm run lint
```
Comment (stu-k) [inline .circleci/config.yml]: This should be a separate job so it's visible in the PR.
https://github.com/silvercar/rac-web-driver/blob/master/.circleci/config.yml#L54
(But don't call it lint-styles, just lint is fine)

**Comment by bucktower** [comment]
@stuartkilgore Decoupled linting check which happens on all commits for good practice and is required to pass for deployment. See the [latest commit linting success](https://circleci.com/gh/silvercar/js-constable/30?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,6 +5,20 @@
...
 
 jobs:
+  lint:
+    working_directory: ~/constable
+    docker:
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/constable
```
Comment (stu-k) [inline .circleci/config.yml]: Is this necessary if no workspace is being persisted in any jobs previously?

File: .circleci/config.yml [hunk 3]
```diff
@@ -18,13 +32,16 @@ jobs:
           command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/constable/.npmrc
       - run:
           name: Publish package
-          command: npm publish --access public  
+          command: npm publish --access public
 
 workflows:
   version: 2
-  build-publish:
+  main:
```
Comment (stu-k) [inline .circleci/config.yml]: `build-lint` maybe? `main` doesn't really mean anything.

**Review by stu-k — APPROVED** [review]
Comments addressed 👍

**Pull request merged**